### PR TITLE
return array when using getImage()

### PR DESCRIPTION
**Pull request by entonbiba**
#1012 #1011

### CODE REVIEW

**Comment by thijstriemstra** [comment]
I would much rather have a data uri with a single canvas and an array with multiple canvasses. That way existing code doesn't break.

**Comment by mspae** [comment]
@entonbiba He means only export an array if there are multiple canvases. Code which relied on the function returning a string and which worked before will still work. (Assuming there weren't multiple canvases)

**Comment by thijstriemstra** [comment]
Come on, it returned a string before, and now it returns an array.

**Comment by entonbiba** [comment]
@thijstriemstra ok lol, updated :)

File: src/drawer.multicanvas.js [hunk 1]
```diff
@@ -337,6 +337,10 @@ WaveSurfer.util.extend(WaveSurfer.Drawer.MultiCanvas, {
             var getEntry = this.canvases[i].wave.getContext('2d');
             availableCanvas.push(getEntry.canvas.toDataURL(type, quality));
         }
-        return availableCanvas;
+        if(this.canvases.length == 1){
```
Comment (thijstriemstra) [inline src/drawer.multicanvas.js]: spaces around parentheses.

File: src/drawer.multicanvas.js [hunk 2]
```diff
@@ -332,11 +332,15 @@ WaveSurfer.util.extend(WaveSurfer.Drawer.MultiCanvas, {
 
     getImage: function(type, quality) {
         // combine all available canvasses together
-        var availableCanvas = '';
+        var availableCanvas = [];
         for (var i in this.canvases) {
             var getEntry = this.canvases[i].wave.getContext('2d');
-            availableCanvas += getEntry.canvas.toDataURL(type, quality);
+            availableCanvas.push(getEntry.canvas.toDataURL(type, quality));
```
Comment (mspae) [inline src/drawer.multicanvas.js]: 1. The `entry` object already includes a cached version of the 2d context.
2. The `canvas` property of the 2d context is just a reference to the underlying canvas HTMLElement. So at the moment this code is unnecessarily complex. Please do instead: `availableCanvas.push(entry.wave.toDataURL(type, quality));` – No need for `getEntry`

File: src/drawer.multicanvas.js [hunk 3]
```diff
@@ -332,11 +332,15 @@ WaveSurfer.util.extend(WaveSurfer.Drawer.MultiCanvas, {
 
     getImage: function(type, quality) {
         // combine all available canvasses together
-        var availableCanvas = '';
+        var availableCanvas = [];
         for (var i in this.canvases) {
```
Comment (mspae) [inline src/drawer.multicanvas.js]: `this.canvases` is an array, please use `this.canvases.forEach(function (entry) {`– This is a pattern which we shouldn't use in this context. see https://github.com/katspaugh/wavesurfer.js/pull/961

File: src/drawer.multicanvas.js [hunk 4]
```diff
@@ -332,11 +332,15 @@ WaveSurfer.util.extend(WaveSurfer.Drawer.MultiCanvas, {
...
             var getEntry = this.canvases[i].wave.getContext('2d');
-            availableCanvas += getEntry.canvas.toDataURL(type, quality);
+            availableCanvas.push(getEntry.canvas.toDataURL(type, quality));
+        }
+        if (this.canvases.length == 1) {
+            return availableCanvas[0];
+        } else {
+            return availableCanvas;
```
Comment (mspae) [inline src/drawer.multicanvas.js]: It's a matter of taste, but I'd prefer if you did `return availableCanvas.length > 1 ? availableCanvas : availableCanvas[0];`

File: src/drawer.multicanvas.js [hunk 5]
```diff
@@ -332,11 +332,15 @@ WaveSurfer.util.extend(WaveSurfer.Drawer.MultiCanvas, {
 
     getImage: function(type, quality) {
```
Comment (mspae) [inline src/drawer.multicanvas.js]: Can you add Jsdoc comments?

**Comment by entonbiba** [comment]
@mspae updated

**Comment by mspae** [comment]
Thanks!

**Pull request merged**